web/client-api/src/main/java/io/deephaven/web/client/api/subscription/ViewportData.java

```diff
@@ -17,6 +17,7 @@
  * Do not assume that the first row in `rows` is the first visible row, because extra rows may be provided for easier
  * scrolling without going to the server.
  */
+// TODO re-add dh.ViewportRow
```

Comment:
TODO - looks like you've still deleted ViewportRow.java

Reply:
Yep, restoring it now, and doing lots of other "make it nice for typescript" to be as compatible as possible with the current version.

Reply:
Ok, the types look to be as good or better than before, though this last commit did add some noise from the Java side.